subprojects/core/src/main/java/org/gradle/api/services/internal/BuildServiceProviderNagger.java

```diff
@@ -51,7 +51,7 @@ private static boolean isServiceRequiredBy(TaskInternal task, BuildServiceProvid
     private static void nagAboutUndeclaredUsageOf(BuildServiceProvider<?, ?> provider, TaskInternal task) {
         deprecateBehaviour(undeclaredBuildServiceUsage(provider, task))
             .withProblemIdDisplayName("Build Service " + provider.getName()+ " undeclared usage")
-            .withAdvice("Declare the association between the task and the build service using 'Task#usesService'.")
+            .withAdvice("Declare the association between the task by declaring the consuming property as a '@ServiceReference'.")
```

Comment:
Indeed, if the task cannot be changed to declare a property to reference the second service, `usesService` allows one to address the issue without requiring a modification to the task definition itself. 

I considered that and decided to omit that advice since it is more of an edge case and we don't want to encourage folks that don't need to use `usesService` to use it.

That would be a workaround for https://github.com/gradle/gradle/issues/24780 btw.

I will look into updating the documentation so it mentions `usesService` as a workaround regarding indirect service usage. If that does not work out, I will update the message above.